### ISSUE TITLE
flaky test: "API nvim_parse_expression works with &opt"

### ISSUE THREAD

**Issue by blueyed** [neovim/neovim#10241]
```
[  ERROR   ] test\unit\viml\expressions\parser_tests.lua @ 6329: API nvim_parse_expression works with &opt
  C:/projects/neovim/test/functional\api\vim_spec.lua:1249: Error while processing test ('&', E):
CUSTOMBUILD : table error :  [C:\projects\neovim\build\functionaltest.vcxproj]
  C:/projects/neovim/test/functional\api\vim_spec.lua:1233: Expected objects to be the same.
  Passed in:
  (table) {
    [1] = 'NvimInvalidOptionSigil:0:0:&'
   *[2] = 'NvimInvalidOptionScope:0:1:'
    [3] = 'NvimInvalidOptionScopeDelimiter:0:2:'
    [4] = 'NvimInvalidOptionName:0:3:' }
  Expected:
  (table) {
    [1] = 'NvimInvalidOptionSigil:0:0:&' }
  
  ({message = 'C:/projects/neovim/test/functional\\api\\vim_spec.lua:1233: Expected objects to be the same.\nPassed in:\n(table) {\n  [1] = \'NvimInvalidOptionSigil:0:0:&\'\n *[2] = \'NvimInvalidOptionScope:0:1:\'\n  [3] = \'NvimInvalidOptionScopeDelimiter:0:2:\'\n  [4] = \'NvimInvalidOptionName:0:3:\' }\nExpected:\n(table) {\n  [1] = \'NvimInvalidOptionSigil:0:0:&\' }'})
  
  stack traceback:
  	C:/projects/neovim/test/functional\api\vim_spec.lua:1249: in function 'check_parsing'
  	test\unit\viml\expressions\parser_tests.lua:6330: in function <test\unit\viml\expressions\parser_tests.lua:6329>
  
  
   53 SKIPPED TESTS
   1 ERROR
  -- Output to stderr:
  
  CMake Error at C:/projects/neovim/cmake/RunTests.cmake:56 (message):
CUSTOMBUILD : functional tests failed with error : 1 [C:\projects\neovim\build\functionaltest.vcxproj]
```
https://ci.appveyor.com/project/neovim/neovim/builds/25272869/job/drahenjssddv3yae#L14884 (MSVC_32)

/cc @ZyX-I

**Comment by justinmk** [comment]
It's a known issue and is blocked waiting on @ZyX-I . I don't think we need a separate ticket for every flaky test.

**Comment by blueyed** [comment]
> I don't think we need a separate ticket for every flaky test.

I think flaky tests should have a high prio / visibility to be adressed/fixed (i.e.should be tracked in issues).

It is annoying to have builds fail due to that, not straight forward to actually look them up in the build logs (folding on Travis gets in the way, the raw log is large / not well structured).
Also it confuses contributors (https://github.com/neovim/neovim/pull/10300#issuecomment-504691617).

> It's a known issue

Discussed elsewhere then already?

**Comment by justinmk** [comment]
No disagreement with any of that, except that I don't think having a bunch of extra tickets will help anything. We can leave this open and see if it helps.

**Comment by blueyed** [comment]
https://ci.appveyor.com/project/neovim/neovim/builds/25470836/job/wuv2ig6794kkpqer (MSVC_32):
```
[  ERROR   ] 1 error, listed below:
  [  ERROR   ] test\unit\viml\expressions\parser_tests.lua @ 6329: API nvim_parse_expression works with &opt
  C:/projects/neovim/test/functional\api\vim_spec.lua:1249: Error while processing test ('(1+&)', E):
CUSTOMBUILD : table error :  [C:\projects\neovim\build\functionaltest.vcxproj]
  C:/projects/neovim/test/functional\api\vim_spec.lua:1233: Expected objects to be the same.
  Passed in:
  (table) {
    [1] = 'NvimNestingParenthesis:0:0:('
    [2] = 'NvimNumber:0:1:1'
    [3] = 'NvimBinaryPlus:0:2:+'
    [4] = 'NvimInvalidOptionSigil:0:3:&'
   *[5] = 'NvimInvalidOptionScope:0:4:)'
    [6] = 'NvimInvalidOptionScopeDelimiter:0:5:'
    [7] = 'NvimInvalidOptionName:0:6:'
    [8] = 'NvimNestingParenthesis:0:4:)' }
  Expected:
  (table) {
    [1] = 'NvimNestingParenthesis:0:0:('
    [2] = 'NvimNumber:0:1:1'
    [3] = 'NvimBinaryPlus:0:2:+'
    [4] = 'NvimInvalidOptionSigil:0:3:&'
   *[5] = 'NvimNestingParenthesis:0:4:)' }
  
  ({message = 'C:/projects/neovim/test/functional\\api\\vim_spec.lua:1233: Expected objects to be the same.\nPassed in:\n(table) {\n  [1] = \'NvimNestingParenthesis:0:0:(\'\n  [2] = \'NvimNumber:0:1:1\'\n  [3] = \'NvimBinaryPlus:0:2:+\'\n  [4] = \'NvimInvalidOptionSigil:0:3:&\'\n *[5] = \'NvimInvalidOptionScope:0:4:)\'\n  [6] = \'NvimInvalidOptionScopeDelimiter:0:5:\'\n  [7] = \'NvimInvalidOptionName:0:6:\'\n  [8] = \'NvimNestingParenthesis:0:4:)\' }\nExpected:\n(table) {\n  [1] = \'NvimNestingParenthesis:0:0:(\'\n  [2] = \'NvimNumber:0:1:1\'\n  [3] = \'NvimBinaryPlus:0:2:+\'\n  [4] = \'NvimInvalidOptionSigil:0:3:&\'\n *[5] = \'NvimNestingParenthesis:0:4:)\' }'})
  
  stack traceback:
  	C:/projects/neovim/test/functional\api\vim_spec.lua:1249: in function 'check_parsing'
  	test\unit\viml\expressions\parser_tests.lua:6475: in function <test\unit\viml\expressions\parser_tests.lua:6329>
```

IIRC I've seen this with MSVC_32 only so far.

**Comment by blueyed** [comment]
This appears to happen on MSVC_32 only.
There are also a lot of warnings there, which might be related: https://ci.appveyor.com/project/neovim/neovim/builds/26489935/job/p1p76if0js8vyamq?fullLog=true#L1261  /cc @ngortheone

**Comment by blueyed** [comment]
Should we `retry` that test?  Would that help, or is it more related to how things have been compiled already?

**Comment by justinmk** [comment]
> Should we `retry` that test? Would that help, or is it more related to how things have been compiled already?

I'd suggest `pending()` or do nothing. `retry()` implies timing-related (i.e. expected) indeterminism, whereas this is more like a memory bug. See also  https://github.com/neovim/neovim/pull/8421#issuecomment-392451423

**Comment by zeertzjq** [comment]
MSVC_32 is no longer supported.
